make/go/dep_golangci_lint.mk

```diff
@@ -3,15 +3,20 @@
 # Must be set
 $(call _assert_var,MAKEGO)
 $(call _conditional_include,$(MAKEGO)/base.mk)
+$(call _conditional_include,$(MAKEGO)/dep_jq.mk)
 $(call _assert_var,CACHE_VERSIONS)
 $(call _assert_var,CACHE_BIN)
 
+# We want to ensure we rebuild golangci-lint every time we require a new Go minor version.
+# Otherwise, the cached version may not support the latest language features.
+GOLANGCI_LINT_GO_VERSION := $(shell go mod edit -json | jq -r .Go | cut -d'.' -f1-2)
```

Comment:
In some downstream projects we specify `go <major>.<minor>.<patch>` to indicate the full toolchain. If we don't mind rebuilding golangci-lint every time that changes, I can remove the `cut` here.